packages/ra-core/src/controller/useSortState.ts

```diff
@@ -16,12 +17,16 @@ const sortReducer = (state: Sort, field: string | Sort): Sort => {
         return field;
     }
     const order =
-        state.field === field && state.order === SORT_ASC
-            ? SORT_DESC
+        state.field === field
+            ? state.order === SORT_ASC
+                ? SORT_DESC
+                : SORT_ASC
             : SORT_ASC;
     return { field, order };
 };
 
+export const defaultSort = { field: 'id', order: 'DESC' };
+
 /**
  * set the sort to the given field, swap the order if the field is the same
  * @name setSort
```

Comment:
the param type is wrong as you can also pass a sort object

Reply:
done